### PR TITLE
CDEV-372/dispense-refill-pickup

### PR DESCRIPTION
**Pull request by pmadria3**
CDEV-372

This PR is a prototype that helps to see for which medications the patient needs to refill.

### CODE REVIEW

**Review by patrickvalle — APPROVED** [review]
lgtm, just non-blocking feedback

File: src/App.tsx [hunk 1]
```diff
@@ -121,6 +122,12 @@ const components: DemoComponent[] = [
   { name: "patient-search", render: () => <PatientSearch />, title: "Patient Search" },
   { name: "timeline", render: () => <PatientTimeline />, title: "Patient Timeline" },
   { name: "encounters", render: () => <PatientEncounters />, title: "Patient Encounters" },
+  {
+    name: "medDispense",
```
Comment (mrosata) [inline src/App.tsx:126]:
```suggestion
    name: "medication-dispense",
```

File: src/components/content/medication-dispense/helpers/dispense-columns.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    },
+    {
+      title: "Performer Details",
+      render: (m) => <div>{m.performerDetails.name}</div>,
+    },
+    {
+      title: "Last Known Fill",
+      render: (m) => <div>{m.whenHandedOver ? m.whenHandedOver : m.whenPrepared}</div>,
+    },
+  ];
```
Comment (mrosata) [inline src/components/content/medication-dispense/helpers/dispense-columns.tsx:22]: Were these in divs for styling? If not we should just stick to
```suggestion
  const dispenseColumns: TableColumn<MedicationDispenseModel>[] = [
    {
      title: "TimeStamp",
      render: (m) => m.alertTimeStamp,
    },
    {
      title: "Medication",
      render: (m) => m.medicationDisplayName,
    },
    {
      title: "Performer Details",
      render: (m) => m.performerDetails.name,
    },
    {
      title: "Last Known Fill",
      render: (m) => m.whenHandedOver ? m.whenHandedOver : m.whenPrepared,
    },
  ];
```

File: src/components/content/medication-dispense/patient-medication-dispense.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import cx from "classnames";
+import { patientMedicationDispenseColumns } from "./helpers/dispense-columns";
+import { ResourceTable } from "../resource/resource-table";
+import { EmptyTable } from "@/components/core/empty-table";
+import { withErrorBoundary } from "@/components/core/error-boundary";
+import { usePatientMedicationDispense } from "@/fhir/medication-dispense";
+
+export type PatientMedDispenseProps = {
+  className?: string;
```
Comment (mrosata) [inline src/components/content/medication-dispense/patient-medication-dispense.tsx:9]:
```suggestion
  className?: cx.Argument;
```

File: src/components/content/medication-dispense/helpers/filters.tsx [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { MedicationDispenseModel } from "@/fhir/models";
+
+export const applyMedDispenseFilters = (data: fhir4.MedicationDispense[]) => {
+  const medDispenseModel = data.map((m) => new MedicationDispenseModel(m));
+
+  return medDispenseModel.filter((m) => m.PMARefillNotPickedUp === "PMARefillNotPickedUp");
+};
```
Comment (mrosata) [inline src/components/content/medication-dispense/helpers/filters.tsx:7]: nit - I'd write this like so, but it's personal preference, PR approved 👍🏼 
```suggestion
export const applyMedDispenseFilters = (data: fhir4.MedicationDispense[]) => data
  .map((m) => new MedicationDispenseModel(m))
  .filter((m) => m.PMARefillNotPickedUp === "PMARefillNotPickedUp");
```

**Review by mrosata — APPROVED** [review]
lgtm, do you need to export the component in index.ts?

**Review by mrosata — DISMISSED** [review]

**Review by mrosata — APPROVED** [review]

**Pull request merged**